lib/truffle/truffle/patching.rb

```diff
@@ -29,7 +40,7 @@ class Module
   private :autoload_without_truffle_patching
 
   def autoload(const, path)
-    if Truffle::Patching::TRUFFLE_PATCHES[path]
+    if Truffle::Patching::TRUFFLE_AFTER_PATCHES[path] || Truffle::Patching::TRUFFLE_BEFORE_PATCHES[path]
       require path
```

Comment:
`autoload` uses directly the internal require (RequireNode), that's Ruby semantics.